src/Umbraco.Core/Auditing/AuditEventHandler.cs

```diff
@@ -0,0 +1,280 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Web;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Models.Membership;
+using Umbraco.Core.Security;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Core.Auditing
+{
+    internal class AuditEventHandler : ApplicationEventHandler
+    {
+        // yada we should inject these
+        private IAuditService AuditServiceInstance => ApplicationContext.Current.Services.AuditService;
+        private IUserService UserServiceInstance => ApplicationContext.Current.Services.UserService;
+
+        private IUser PerformingUser
+        {
+            get
+            {
+                var identity = Thread.CurrentPrincipal?.GetUmbracoIdentity();
+                return identity == null
+                    ? new User { Id = 0, Name = "(no user)", Email = "" }
+                    : UserServiceInstance.GetUserById(Convert.ToInt32(identity.Id));
+            }
+        }
+
+        private string PerformingIp
+        {
+            get
+            {
+                var httpContext = HttpContext.Current == null ? (HttpContextBase) null : new HttpContextWrapper(HttpContext.Current);
+                var ip = httpContext.GetCurrentRequestIpAddress();
+                if (ip.ToLowerInvariant().StartsWith("unknown")) ip = "";
+                return ip;
+            }
+        }
+
+        protected override void ApplicationStarted(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
+        {
+            //BackOfficeUserManager.AccountLocked += ;
```

Comment:
Any reason we don't track these commented out bits?

Reply:
Currently logging what I felt made sense, GDPR-wise. Left these in comments so we can have the discussion. Methink we don't need them. You?